rustls-libssl/tests/runner.rs

```diff
@@ -306,3 +372,20 @@ fn wait_for_port(port: u16) -> Option<()> {
         }
     }
 }
+
+fn wait_for_stdout(stream: &mut Child, expected: &[u8]) {
+    let stdout = stream.stdout.as_mut().unwrap();
+
+    let mut buffer = Vec::with_capacity(1024);
+
+    loop {
+        let mut input = [0u8];
+        let new = stdout.read(&mut input).unwrap();
+        assert_eq!(new, 1);
+        buffer.push(input[0]);
```

Comment:
Can you add a brief comment explaining why it's important to read char by char here? It seems like a deliberate choice but I'm not sure I see the rationale.

Reply:
Added a doc comment to this function. If we read in large chunks, we would have to do two extra things:

- not use `ends_with`, instead make our own searching iterator using `windows()` over the newly-read data. annoying.
- return the excess (after the `expected` string) to the caller, and then accept a previous excess buffer that is search through before reading further data from stdout (and also glue this to previous output when logging in `print_output`). In C you could use `ungetc`.